src/AutoMapper/TypeMap.cs

```diff
@@ -184,7 +184,7 @@ public bool PassesCtorValidation()
 
             var constructors = DestinationTypeToUse
                 .GetDeclaredConstructors()
-                .Where(ci => !ci.IsStatic);
+                .Where(ci => !ci.IsStatic && Profile.ShouldMapCtor(ci));
```

Comment:
This should just check Constructors in TypeDetails.

Reply:
It should have done that in the first place too 😄 I'll change it.